### PR TITLE
[WIP] Remove enrollment track

### PR DESCRIPTION
**Pull request by dyudyunov**
YT: https://youtrack.raccoongang.com/issue/OeX_Depr-185

Completely remove enrollment_track functionality.
Testing should consider checking:
- masquerading user for upselling link
(https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/courseware/utils.py#L59)
- `can_show_streak_discount_coupon` functionality
(https://github.com/openedx/edx-platform/pull/27505/files#r628294234)
- course masquerade in general

Looks like content type gating decision docs should be updated:
https://github.com/openedx/edx-platform/tree/master/openedx/features/content_type_gating/docs/decisions

Enrollment_track documentation should be removed too:
https://github.com/search?q=org%3Aopenedx+enroll_track_courseware&type=code

### CODE REVIEW

**Review by ruzniaievdm — APPROVED** [review]

File: openedx/core/djangoapps/courseware_api/views.py [hunk 1]
```diff
@@ -346,18 +337,12 @@ def user_timezone(self):
 
     @property
     def can_access_proctored_exams(self):
-        """Returns if the user is eligible to access proctored exams"""
-        if is_masquerading_as_non_audit_enrollment(
```
Comment (idegtiarov) [inline openedx/core/djangoapps/courseware_api/views.py:350]: Now staff could masquerade as audit user, why do that functionality is removed?

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Some conceptual questions are inline, please take a look. Other LGTM!

File: lms/djangoapps/courseware/utils.py [hunk 2]
```diff
@@ -52,56 +49,17 @@ def verified_upgrade_deadline_link(user, course=None, course_id=None):
     return EcommerceService().upgrade_url(user, course_id)
 
 
-def is_mode_upsellable(user, enrollment, course=None):
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply lms/djangoapps/courseware/utils.py:55]: No, the course could still be upgraded to verified mode, I returned this functionality to the state close to one it was before enrollment_track was introduced. Now it checks the enrollment mode directly, not through the user partition. Check this out for example https://github.com/raccoongang/edx-platform/pull/2430/files#diff-7da1f29a8bd9ddecbf10fb284dd508a6ee70ff1322d4475782c7a6e0815c932fR111

File: openedx/core/djangoapps/courseware_api/views.py (hunk 1 above)
Comment (dyudyunov) [reply openedx/core/djangoapps/courseware_api/views.py:350]: The logic stays the same here. Masquerade for a particular mode is not available anymore (that was a part of enrollment_track partition), so we just check the enrollment type

File: openedx/core/djangoapps/django_comment_common/migrations/0010_alter_coursediscussionsettings_division_scheme.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.13 on 2022-05-23 08:12
```
[the remark this answers is not in the document]
Comment (dyudyunov) [reply openedx/core/djangoapps/django_comment_common/migrations/0010_alter_coursediscussionsettings_division_scheme.py:1]: I'll double-check the doc and return with an answer, but for me it looks appropriate

**Pull request closed without merging**